components/installer/pkg/config/installation-data.go

```diff
@@ -95,12 +93,11 @@ func (installationData *InstallationData) ShouldInstallComponent(componentName s
 	return found
 }
 
-func convertToMap(cList string) map[string]struct{} {
-	split := strings.Split(strings.Replace(cList, " ", "", -1), ",")
-	output := make(map[string]struct{}, len(split))
-	for _, c := range split {
-		if c != "" {
-			output[c] = struct{}{}
+func convertToMap(cList []v1alpha1.KymaComponent) map[string]v1alpha1.KymaComponent {
+	output := make(map[string]v1alpha1.KymaComponent, len(cList))
+	for _, c := range cList {
+		if c.Name != "" {
+			output[c.Name] = c
```

Comment:
is it possible that this list contains duplicates? What in such case?

Reply:
For now it is not a problem, because installation process is still hardcoded so every component will be installed only once. In further changes this needs to be addressed.